src/mon/MonMap.h

```diff
@@ -25,34 +25,124 @@ namespace ceph {
   class Formatter;
 }
 
+struct mon_info_t {
+  /**
+   * monitor name
+   *
+   * i.e., 'foo' in 'mon.foo'
+   */
+  string name;
+  /**
+   * monitor's public address
+   *
+   * public facing address, traditionally used to communicate with all clients
+   * and other monitors.
+   */
+  entity_addr_t public_addr;
```

Comment:
We could include a 'weight' value here so that you can make clients prefer certain mons?

We could also include a map<string,string> loc (that matches the crush hierarchy) so that you can prefer mons in the same rack or data center.